File: glip-team-messaging/calendar-events/create-event-by-group-id/code-samples/createEventbyGroupId.py
# https://developers.ringcentral.com/my-account.html#/applications
# Find your credentials at the above url, set them as environment variables, or enter them below

# PATH PARAMETERS
groupId = '<ENTER VALUE>'

# POST BODY
body = {
    'id': '<ENTER VALUE>',
    'creatorId': '<ENTER VALUE>',
    'title': '<ENTER VALUE>',
    'startTime': '<ENTER VALUE>',
    'endTime': '<ENTER VALUE>',
    'allDay': true,
    'recurrence': 'None',
    'endingCondition': '<ENTER VALUE>',
    'endingAfter': 000,
    'endingOn': 'None',
    'color': 'Black',
    'location': '<ENTER VALUE>',
    'description': '<ENTER VALUE>'
}

import os
from ringcentral import SDK
rcsdk = SDK(os.environ['clientId'], os.environ['clientSecret'], os.environ['serverURL'])
platform = rcsdk.platform()
platform.login(os.environ['username'], os.environ['extension'], os.environ['password'])
r = platform.post(f'/restapi/v1.0/glip/groups/{groupId}/events', body)

File: glip-team-messaging/calendar-events/create-event-by-group-id/code-samples/createEventbyGroupId.php
<?php
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
$groupId = '<ENTER VALUE>';

// POST BODY
$body = array(
    'id' => '<ENTER VALUE>',
    'creatorId' => '<ENTER VALUE>',
    'title' => '<ENTER VALUE>',
    'startTime' => '<ENTER VALUE>',
    'endTime' => '<ENTER VALUE>',
    'allDay' => true,
    'recurrence' => 'None',
    'endingCondition' => '<ENTER VALUE>',
    'endingAfter' => 000,
    'endingOn' => 'None',
    'color' => 'Black',
    'location' => '<ENTER VALUE>',
    'description' => '<ENTER VALUE>'
)

require('vendor/autoload.php');
$rcsdk = new RingCentral\SDK\SDK(getenv('clientId'), getenv('clientSecret'), getenv('serverURL'));
$platform = $rcsdk->platform();
$platform->login(getenv('username'), getenv('extension'), getenv('password'));
$r = $platform->post("/restapi/v1.0/glip/groups/{$groupId}/events", $body);
?>
